catchup/networkFetcher.go

```diff
@@ -92,20 +94,36 @@ func (netFetcher *NetworkFetcher) FetchBlock(ctx context.Context, round basics.R
 			}
 			netFetcher.log.Infof("FetchBlock: failed to download block %d on attempt %d out of %d. %v", round, retryCount, netFetcher.cfg.CatchupBlockDownloadRetryAttempts, err)
 			netFetcher.peerSelector.rankPeer(psp, peerRankDownloadFailed)
-		} else if !blk.ContentsMatchHeader() && blk.Round() > 0 {
+			continue // retry the fetch
+		}
+
+		// Check that the block's contents match the block header
+		if !blk.ContentsMatchHeader() && blk.Round() > 0 {
 			netFetcher.peerSelector.rankPeer(psp, peerRankInvalidDownload)
 			// Check if this mismatch is due to an unsupported protocol version
 			if _, ok := config.Consensus[blk.BlockHeader.CurrentProtocol]; !ok {
 				netFetcher.log.Errorf("FetchBlock: downloaded block(%v) unsupported protocol version detected: '%v'", round, blk.BlockHeader.CurrentProtocol)
 			}
 			netFetcher.log.Warnf("FetchBlock: downloaded block(%v) contents do not match header", round)
 			netFetcher.log.Infof("FetchBlock: failed to download block %d on attempt %d out of %d. %v", round, retryCount, netFetcher.cfg.CatchupBlockDownloadRetryAttempts, err)
-		} else {
-			// upon successful download rank the peer according to the download speed
-			peerRank := netFetcher.peerSelector.peerDownloadDurationToRank(psp, downloadDuration)
-			netFetcher.peerSelector.rankPeer(psp, peerRank)
-			return blk, cert, downloadDuration, err
+			continue // retry the fetch
+		}
+
+		// Authenticate the block. for correct execution, caller should call FetchBlock only when the lookback block is available
+		if netFetcher.cfg.CatchupVerifyCertificate() {
+			err = netFetcher.auth.Authenticate(blk, cert)
+			if err != nil {
+				netFetcher.log.Warnf("FetchBlock: cert did not authenticate block %d on attempt %d out of %d. %v", round, retryCount, netFetcher.cfg.CatchupBlockDownloadRetryAttempts, err)
```

Comment:
Here the text of the error message suggest `did not authenticate`. In fact, it authenticated but failed.
Maybe say 'cert authentication failed`.